src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,59 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+type Props = {
+  searchQuery: string;
+  setSearchQuery: (searchQuery: string) => void;
+  status: string;
+  setStatus: (status: string) => void;
+};
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = (props) => {
+  const {
+    searchQuery,
+    setSearchQuery,
+    status,
+    setStatus,
+  } = props;
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            value={status}
+            data-cy="statusSelect"
+            onChange={event => setStatus(event.target.value)}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={searchQuery}
+          onChange={event => setSearchQuery(event.target.value)}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {searchQuery && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
Try to add  attribute to button to disable the linter.
I will attach a link to options for how to do it:
https://stackoverflow.com/questions/57616720/react-jsx-a11y-eslint-throws-unexpected-errors-for-control-label